### ISSUE TITLE
Non-chapel nightly tests run on chapNN slaves fail w unknown pip install errors (Cray int Jenkins)

### ISSUE THREAD

**Issue by awallace-cray** [chapel-lang/chapel#8871]
Four nightly Jenkins test jobs which run on chapNN slaves, test various things other than Chapel itself (pychapel for example), and which use a shiningpanda Python virtual env provided by Jenkins, started failing intermittently on 03/17 w unknown pip install errors. 

The four Jenkins jobs are 
correctness-test-pychapel
correctness-test-github-email-notifications
correctness-test-sphinxcontrib-chapeldomain
correctness-test-chplforpyp-docs

Typical error below (details of the stack trace may vary, this is from correctness-test-sphinxcontrib-chapeldomain on chap16): 
```
+ pip install -r requirements.txt -r test-requirements.txt -r docs-requirements.txt
Collecting docutils (from -r requirements.txt (line 1))
Exception:
Traceback (most recent call last):
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/commands/install.py", line 335, in run
    wb.build(autobuilding=True)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/wheel.py", line 749, in build
    self.requirement_set.prepare_files(self.finder)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/req/req_set.py", line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/req/req_set.py", line 554, in _prepare_file
    require_hashes
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/req/req_install.py", line 278, in populate_link
    self.link = finder.find_requirement(self, upgrade)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/index.py", line 465, in find_requirement
    all_candidates = self.find_all_candidates(req.name)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/index.py", line 423, in find_all_candidates
    for page in self._get_pages(url_locations, project_name):
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/index.py", line 568, in _get_pages
    page = self._get_page(location)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/index.py", line 683, in _get_page
    return HTMLPage.get_page(link, session=self.session)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/index.py", line 792, in get_page
    "Cache-Control": "max-age=600",
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/_vendor/requests/sessions.py", line 521, in get
    return self.request('GET', url, **kwargs)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/download.py", line 386, in request
    return super(PipSession, self).request(method, url, *args, **kwargs)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/_vendor/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/_vendor/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/_vendor/cachecontrol/adapter.py", line 37, in send
    cached_response = self.controller.cached_request(request)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/_vendor/cachecontrol/controller.py", line 111, in cached_request
    resp = self.serializer.loads(request, cache_data)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/_vendor/cachecontrol/serialize.py", line 114, in loads
    return getattr(self, "_loads_v{0}".format(ver))(request, data)
  File "/ptmp/jenkins/chapel-ci.bogus/shiningpanda/jobs/2198da4f/virtualenvs/d41d8cd9/lib/python2.7/site-packages/pip/_vendor/cachecontrol/serialize.py", line 168, in _loads_v1
    cached = pickle.loads(data)
ImportError: No module named urllib3._collections
Build step 'Virtualenv Builder' marked build as failure
```

**Comment by awallace-cray** [comment]
Note: these four jobs were already using the same hand-built Python 2.7 described in #8866, and had been doing so for a long time.  However, in these four jobs, the virtual env was created by Jenkins Shiningpanda plugin instead of a Chapel `make *-venv` command. 

All four of these jobs could be made to work again by rm'ing the local Pip download cache (in a directory given by `XDG_CACHE_HOME`) and re-running.  On 03/17, I thought doing that once, on each chapNN machine, was all that was needed. 

However, on 03/18, the latest nightly builds of these jobs had failed again. 

Once any Chapel smoke / correctness test affected by #8866 had run on a given chapNN machine, any of these four jobs failed (with the above pip install errors) the next time it happened to run on the same machine. 

It appears the Chapel-created Python virtual env leaves something in the local Pip download cache, which causes these four non-Chapel test jobs to fail with the above pip install errors.  
Interestingly, this was true even for chap05, which is a recently-built SLES-12 system (all other chapNN are old, SLES-11 systems). 

The changes made to the Chapel smoke / correctness jobs (#8866) probably caused Pip / Python to try and share more cache contents with these non-Chapel / shiningpanda venv jobs (on the same machine), because after #8866 was fixed, both sets of jobs were running Python 2.7.  (Before that fix, the Chapel smoke / correctness tests used Python 2.6.)

**Comment by awallace-cray** [comment]
Note: pip install option `--no-cache-dir` may have fixed this. 
It was not tested. 
Also, I was reluctant to make these test jobs actually download many MB of data from pypi.com every night.
